test/Interpreter/generic_casts.swift

```diff
@@ -157,27 +158,104 @@ func genericAnyIs<T>(type: Any.Type, to: T.Type, expected: Bool) -> Bool {
     return expected
   }
 }
+// `as?` checks
+func nongenericAnyAsConditionalP(type: Any.Type) -> Bool {
+  return (type as? P.Protocol) != nil
+}
+func nongenericAnyAsConditionalPAndAnyObject(type: Any.Type) -> Bool {
+  return (type as? (P & AnyObject).Protocol) != nil
+}
+func nongenericAnyAsConditionalPAndPCSub(type: Any.Type) -> Bool {
+  return (type as? (P & PCSub).Type) != nil
+}
+func genericAnyAsConditional<T>(type: Any.Type, to: T.Type, expected: Bool) -> Bool {
+  // If we're testing against a runtime that doesn't have the fix this tests,
+  // just pretend we got it right.
+  if #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) {
+    return (type as? T.Type) != nil
+  } else {
+    return expected
+  }
+}
+// `as!` checks
+func blackhole<T>(_ : T) { }
+
+func nongenericAnyAsUnconditionalP(type: Any.Type) -> Bool {
+  blackhole(type as! P.Protocol)
+  return true
+}
+func nongenericAnyAsUnconditionalPAndAnyObject(type: Any.Type) -> Bool {
+  blackhole(type as! (P & AnyObject).Protocol)
+  return true
+}
+func nongenericAnyAsUnconditionalPAndPCSub(type: Any.Type) -> Bool {
+  blackhole(type as! (P & PCSub).Type)
+  return true
+}
+func genericAnyAsUnconditional<T>(type: Any.Type, to: T.Type, expected: Bool) -> Bool {
+  if #available(macOS 9999, iOS 9999, watchOS 9999, tvOS 9999, *) {
+    blackhole(type as! T.Type)
+  }
+  return true
+}
+
 // CHECK-LABEL: casting types to protocols with generics:
 print("casting types to protocols with generics:")
-print(nongenericAnyIsP(type: PS.self)) // CHECK: true
+print(nongenericAnyIsP(type: P.self)) // CHECK: true
+print(genericAnyIs(type: P.self, to: P.self, expected: true)) // CHECK: true
+print(nongenericAnyIsP(type: PS.self)) // CHECK-ONONE: true
 print(genericAnyIs(type: PS.self, to: P.self, expected: true)) // CHECK-ONONE: true
-print(nongenericAnyIsP(type: PE.self)) // CHECK: true
+print(nongenericAnyIsP(type: PE.self)) // CHECK-ONONE: true
```

Comment:
These existing checks should not have changed.  I need to dig a little more to figure out what's going on here.